astropy/wcs/low_level_api.py

```diff
@@ -0,0 +1,159 @@
+import abc
+
+import numpy as np
+
+__all__ = ['BaseLowLevelWCS']
+
+
+class BaseLowLevelWCS(metaclass=abc.ABCMeta):
+    """
+    Abstract base class for the low-level WCS interface described in APE 14
+    (https://doi.org/10.5281/zenodo.1188875)
+    """
+
+    @abc.abstractproperty
+    def pixel_n_dim(self):
+        """
+        The number of axes in the pixel coordinate system.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractproperty
+    def world_n_dim(self):
+        """
+        The number of axes in the world coordinate system.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractproperty
+    def world_axis_physical_types(self):
+        """
+        An iterable of strings describing the physical type for each world axis.
+
+        These should be names from the VO UCD1+ controlled Vocabulary
+        (http://www.ivoa.net/documents/latest/UCDlist.html). If no matching UCD
+        type exists, this can instead be ``"custom:xxx"``, where ``xxx`` is an
+        arbitrary string.  Alternatively, if the physical type is
+        unknown/undefined, an element can be `None`.
+        """
+        raise NotImplementedError()
```

Comment:
I wonder if this base class should also provide validation for the vocabulary?  E.g., there could be a line in the docs like do 
```
self._validate_physical_types(toreturn)
return toreturn
```
And that checks that the vocabulary is matched?

Whether this is worth doing depends on how hard it is to get a canonical list of the vocab, though...